### PR TITLE
Dynamic data & cleanup

### PR DESCRIPTION
**Pull request by sadiecrawford**
I'd love your opinion on how the categories and icons are now being dynamically generated, rather than just being hard-coded as before. For instance, should I be using a separate file (e.g. json or similar) with this info?

### CODE REVIEW

File: src/components/portfolio_elements/PortfolioPreview.vue [hunk 1]
```diff
@@ -1,22 +1,33 @@
...
   name: 'PortfolioPreview',
   props: {
     imageSrc: {
       type: String,
       required: true
     }
+  },
+  methods: {
+    getPreviewImageUrl(imagePath: string): string {
+      return new URL('/src/assets/images/portfolio/' + imagePath, import.meta.url).href
```
Comment (ufo-catcher) [inline src/components/portfolio_elements/PortfolioPreview.vue]: Love this but let's move the path to an env variable or the like so you can maintain it outside of the component.

Comment (ufo-catcher) [reply src/components/portfolio_elements/PortfolioPreview.vue]: For instance, you could define it as an alias in the Vite config.

File: src/views/HomeView.vue [hunk 2]
```diff
@@ -1,35 +1,88 @@
...
+export default {
+  data() {
+    return {
+      categories: [
+        {
+          name: 'Games',
+          items: [
+            {
+              title: 'Unmasked',
+              previewImg: 'games/unmasked/UnmaskedPreview.png',
```
Comment (ufo-catcher) [inline src/views/HomeView.vue]: Same comment here about having a central source of truth for paths. Maybe worth making a helper function or component that encapsulates all of the logic surrounding finding images. You could even do it via an enum or a key/value data struct.

File: src/views/HomeView.vue [hunk 3]
```diff
@@ -1,35 +1,88 @@
...
+      </div>
+    </VerticalBox>
+  </main>
 </template>
+
+<script lang="ts">
+export default {
+  data() {
+    return {
+      categories: [
```
Comment (ufo-catcher) [inline src/views/HomeView.vue:42]: Yes, excellent, I much prefer iterating on data to hardcoding markup.

**Review by ufo-catcher — CHANGES_REQUESTED** [review]
Let's figure out how to handle aliasing or centralizing paths and then this is good to go!

File: src/components/portfolio_elements/PortfolioPreview.vue [hunk 4]
```diff
@@ -1,22 +1,64 @@
...
       type: String,
       required: true
+    },
+    imageDesc: {
+      type: String,
+      required: false
+    }
+  },
+  methods: {
+    getPreviewImageUrl(categoryStr: String, imagePath: string): string {
```
Comment (ufo-catcher) [inline src/components/portfolio_elements/PortfolioPreview.vue]: I think we should make this its own helper function/export and break it out of this component for use elsewhere!

File: src/components/portfolio_elements/PortfolioPreview.vue [hunk 5]
```diff
@@ -1,22 +1,64 @@
...
+          break
+        case Category.Software:
+          root = import.meta.env.VITE_SOFTWARE_URL
+          break
+        default:
+          // default is already set above
+          break
+      }
+
+      return `${root}${imagePath}`
```
Comment (ufo-catcher) [inline src/components/portfolio_elements/PortfolioPreview.vue]: We can make this part stateless by returning early in the switch statement and not using let.

File: src/views/HomeView.vue [hunk 6]
```diff
@@ -1,35 +1,99 @@
...
-    <PortfolioSection>
-      <PortfolioPreview :imageSrc="img_UnmaskedPreview" />
-      <PortfolioPreview :imageSrc="img_HushPreview" />
-    </PortfolioSection>
+  <main>
+    <VerticalBox>
+      <h1 class="center-wrapper">Work.</h1>
+      <HorizontalBox class="center-wrapper">
+        <div v-for="(category, index) in categories" :key="index" class="category">
+          <SimpleLink :href="'#' + category.name">{{ category.name }}</SimpleLink>
```
Comment (ufo-catcher) [inline src/views/HomeView.vue]: Was there a reason to use this kind of concatenation here?

File: src/views/HomeView.vue [hunk 7]
```diff
@@ -1,35 +1,99 @@
...
-      <PortfolioPreview :imageSrc="img_DentaLabPreview" />
-    </PortfolioSection>
-    <h2 class="left-wrapper">Software.</h2>
-  </VerticalBox>
+      <div v-for="(category, index) in categories" :key="index" class="category">
+        <h2 :id="category.name" class="left-wrapper" style="scroll-margin-top: 96px">
+          {{ category.name }}
+        </h2>
+        <PortfolioSection>
+          <div v-for="(item, itemIndex) in category.items" :key="itemIndex" class="item">
```
Comment (ufo-catcher) [inline src/views/HomeView.vue]: btw you are not required to expose the second param in v-for loops if you don't use it (not that it matters really)

**Review by ufo-catcher — APPROVED** [review]
Real good improvements, nice!

File: src/views/HomeView.vue (hunk 7 above)
Comment (sadiecrawford) [reply src/views/HomeView.vue]: Ah, that's good to know though! I'd prefer to hide it if I don't use it

File: src/views/HomeView.vue (hunk 6 above)
Comment (sadiecrawford) [reply src/views/HomeView.vue]: Habit lol

File: src/components/portfolio_elements/PortfolioPreview.vue (hunk 4 above)
Comment (sadiecrawford) [reply src/components/portfolio_elements/PortfolioPreview.vue]: Good call!

**Pull request merged**